File: client/MauiChat/Models/ContactDto.cs
using CommunityToolkit.Mvvm.ComponentModel;

namespace MauiChat.Models;

/// <summary>
/// Matches server <c>ContactDto</c>:
/// id, name, avatarUrl, isOnline, lastMessage, lastMessageAt, isGroup.
///
/// <see cref="Id"/> is a string so it can hold either a friend's user-id
/// (Guid) for a private chat, or a group-id for a group chat.
///
/// Implemented as <see cref="ObservableObject"/> because <see cref="IsOnline"/>
/// is updated in-place from live SignalR presence events.
/// </summary>
public partial class ContactDto : ObservableObject
{
    [ObservableProperty] private string _id = string.Empty;
    [ObservableProperty] private string _name = string.Empty;
    [ObservableProperty] private string? _avatarUrl;
    [ObservableProperty] private bool _isOnline;
    [ObservableProperty] private string? _lastMessage;
    [ObservableProperty] private DateTime? _lastMessageAt;
    [ObservableProperty] private bool _isGroup;
}

File: client/flutter_chat/lib/models/contact_dto.dart
class ContactDto {
  final String id;
  final String name;
  final String? avatarUrl;
  final bool isOnline;
  final String? lastMessage;
  final DateTime? lastMessageAt;
  final bool isGroup;

  const ContactDto({
    required this.id,
    required this.name,
    this.avatarUrl,
    required this.isOnline,
    this.lastMessage,
    this.lastMessageAt,
    required this.isGroup,
  });

  factory ContactDto.fromJson(Map<String, dynamic> j) => ContactDto(
        id: j['id'] as String,
        name: (j['name'] as String?) ?? '',
        avatarUrl: j['avatarUrl'] as String?,
        isOnline: j['isOnline'] as bool? ?? false,
        lastMessage: j['lastMessage'] as String?,
        lastMessageAt: j['lastMessageAt'] == null
            ? null
            : DateTime.parse(j['lastMessageAt'] as String),
        isGroup: j['isGroup'] as bool? ?? false,
      );
}
